e2e-tests/src/main/java/uk/nhs/adaptors/gp2gp/Mongo.java

```diff
@@ -2,22 +2,46 @@
 
 import static com.mongodb.client.MongoClients.create;
 
+import java.time.Instant;
+
+import org.bson.Document;
+
 import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
 import com.mongodb.client.model.Filters;
-import org.bson.Document;
+import com.mongodb.client.model.Updates;
 
 public class Mongo {
-
     public static MongoCollection<Document> getCollection() {
-        var connectionString = System.getenv().getOrDefault("GP2GP_MONGO_URI", "mongodb://localhost:27017");
-        var database = System.getenv().getOrDefault("GP2GP_MONGO_DATABASE_NAME", "gp2gp");
-        var client = create(connectionString);
-        var db = client.getDatabase(database);
-        return db.getCollection("ehrExtractStatus");
+        return prepareDatabaseConnection().getCollection("ehrExtractStatus");
     }
 
     public static Document findEhrExtractStatus(String conversationId) {
         var collection = getCollection();
         return collection.find(Filters.eq("conversationId", conversationId)).first();
     }
+
+    public static Document findEhrExtractStatusWithGpcAccessDocument(String conversationId) {
+        var collection = getCollection();
+        return collection.find(Filters.and(Filters.eq("conversationId", conversationId), 
+            Filters.size("gpcAccessDocuments", 1)))
+            .first();
+    }
+
+    public static void addAccessDocument(String conversationId, String documentId, String taskId, Instant accessedAt) {
+        Document document = new Document();
+        document.append("objectName", documentId + ".json");
+        document.append("taskId", taskId);
```

Comment:
The test should only need to pre-populate "objectName" and should assert that "accessedAt" and "taskId" have been updated right?

Reply:
Ok

Reply:
These were the additional meta data needed, so I think we need these test?